discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -107,11 +108,27 @@ def query_ipfs_metadata_json(self, gateway_ipfs_urls, metadata_format):
                     logger.error(f"IPFSClient | {url} generated an exception: {exc}")
         return formatted_json
 
-    def get_metadata_from_gateway(self, multihash, metadata_format):
+    def get_metadata_from_gateway(self, multihash, metadata_format, user_replica_set=None):
         # Default return initial metadata format
         gateway_metadata_json = metadata_format
-        logger.warning(f"IPFSCLIENT | get_metadata_from_gateway, {multihash}")
+        logger.warning(f"IPFSCLIENT | get_metadata_from_gateway, {multihash} replica set: {user_replica_set}")
         gateway_endpoints = self._cnode_endpoints
+
+        # first attempt to first fetch metadata from user replica set, if provided & non-empty
+        if (user_replica_set and isinstance(user_replica_set, str)):
+            user_replicas = user_replica_set.split(",")
+            try:
+                query_urls = ["%s/ipfs/%s" % (addr, multihash) for addr in user_replicas]
+                data = self.query_ipfs_metadata_json(query_urls, metadata_format)
```

Comment:
maybe not in this PR, but I'm a little worried that the 1s timeout introduces a possibility of indexing getting halted if everything is slow to respond (under extreme load). Maybe we could have this cycle through with 1s timeouts and if that fails, do an extended timeout retry?

Reply:
agreed, made card https://linear.app/audius/issue/AUD-502/monitor-ipfscat-indexing-consider-adding-a-larger-timeout-on-retry